app/services/wakes/all_pageviews_count_updater_service.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+require 'google/apis/analytics_v3'
+
+class Wakes::AllPageviewsCountUpdaterService
+  def initialize(start_year, end_year = current_year)
+    @start_year = start_year
+    @end_year = end_year
+  end
+
+  def update_path_counts
+    date_ranges.each do |date_range|
+      GetPathCountsForDateRangeService.new(date_range).path_counts.each do |path, count|
+        update_path_count(path, count)
+      end
+    end
+  end
+
+  private
+
+  attr_reader :start_year, :end_year
+
+  def current_year
+    Date.today.year
+  end
+
+  def date_ranges
+    @date_ranges ||= begin
+      start_date = Date.new(start_year, 1, 1)
+      end_date = current_year == end_year ? 1.day.ago.to_date : Date.new(end_year, 12, 31)
+      DateRanges.new(start_date..end_date, :year)
+    end
+  end
+
+  def update_path_count(path, count)
+    location = Wakes::Location.where(:path => path).first
+    return unless location.present?
+
+    location.pageview_counts[@year] = count
+    location.save
+    location.resource&.update_pageview_count
+    begin
+      location.resource.wakeable&.update_pageview_count
+    rescue NameError => ex # Invalid wakeable
+      puts ex.message
+    end
+  end
+
+  class DateRanges
+    include Enumerable
+
+    def initialize(date_range, range)
```

Comment:
not really liking the `range` variable here. It feels like a `type` or `field` or something similar would be better.